### PR TITLE
fix(lambda-nodejs): parcel is too big to bundle

### LINKED ISSUE #6340: lambda-nodejs: parcel is too big to bundle
We are looking into releasing the entire AWS CDK as a single library (see https://github.com/aws/aws-cdk-rfcs/issues/6). This basically means that all modules (and their transitive dependencies) will be bundled into a single package.

Recently, we have introduced @aws-cdk/aws-lambda-nodejs as a high level API for writing node.js lambda functions. This construct uses [parcel] in order to bundle all function's dependencies into a single artifact.

The problem is that [parcel] is too big:
- It has [652](http://npm.broofa.com/?q=parcel) transitive dependencies
- 15,558 files
- 100MiB (uncompressed)

This is unrealistic and also introduces an intrinsic security risk posed by including so many dependencies in our project.

### Proposal

Let's use parcel as an [undeclared] peer-dependency. Simply require that users declare parcel in their own `package.json` file. It will also give them more control over which version to use.

### PR DESCRIPTION
**Pull request by jogold**
Stop bundling `parcel-bundler` and make it an undeclared peer dependency.

Transform `build` function into a class.

Fixes #6340

BREAKING CHANGE: `parcel-bundler` v1.x is now a peer dependency of `@aws-cdk/aws-lambda-nodejs`.
Please add it to your `package.json`.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-lambda-nodejs/README.md [hunk 1]
```diff
@@ -17,6 +17,8 @@
 
 This library provides constructs for Node.js Lambda functions.
 
+⚠️ This module has a peer dependency on [`parcel-bundler`](https://www.npmjs.com/package/parcel-bundler) v1.x.
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda-nodejs/README.md]: I think this requires a bit more details. Something like this:

---

To use this module, you will need to add a dependency on `parcel-bundler` in your `package.json`:

```shell
yarn add parcel-bundler@^1.0.0
# or
npm install parcel-bundler@^1.0.0
```

File: packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+/**
+ * Builder
+ */
+export class Builder {
+  private readonly parcelBinPath: string;
+
+  constructor(private readonly options: BuilderOptions) {
+    const parcelPkgPath = require.resolve('parcel-bundler/package.json'); // This will throw if `parcel-bundler` cannot be found
```
Comment (eladb) [inline packages/@aws-cdk/aws-lambda-nodejs/lib/builder.ts]: Probably want to `try-catch` and throw a nice error explaining that parcel needs to be installed

**Comment by jogold** [comment]
On it

**Comment by eladb** [comment]
Beautiful, thanks!

**Comment by jogold** [comment]
@eladb can you squash this manually so that the BREAKING CHANGE is included at the bottom of the commit message?

**Pull request merged**